pkg/core/state/client.go

```diff
@@ -26,6 +27,8 @@ type Client struct {
 	dsn    string
 	db     *sdkdb.DB
 	Logger hclog.Logger
+	// StoreRunResults indicates whether to persist a run result
+	StoreRunResults bool
```

Comment:
Having a bool here to enable/disable some functionality in client seems weird to me as a pattern

Reply:
For the time being since the feature is opt-in and on by default it's way simpler to store the feature flag there and not to pass it down all the way in order to toggle the client calls. In my previous implementation before I rebased the state manager it was all in one place.

Reply:
@disq though about it, for further features it already makes sense to pass the flag down all the way and not in the client. It's addressed in the latest version